example/ios/AdyenExampleTests/Modules/ApplePayModuleTests.swift

```diff
@@ -0,0 +1,208 @@
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testIsAvailableReturnFalseWhenNoAmount() throws {
+      // Given
+    mockPaymentAuthorizationService.canMakePaymentsResult = false
+    let sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+    let configuration: NSDictionary = [
+      "clientKey": "test_client_key",
+      "countryCode": "US",
+      "applepay": [
+        "merchantID": "merchant.com.adyen.test",
+        "merchantName": "Test Merchant"
+      ]
+    ]
+
+      // Then
+    sut.isAvailable(paymentMethodDict, configuration: configuration) { resolve in
+      let isSuccess = try! XCTUnwrap(resolve as? Bool )
+      XCTAssertFalse(isSuccess)
+    } rejecter: { title, message, error in
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testIsAvailableReturnFalseWhenNoViewController() throws {
+      // Given
+    mockPaymentAuthorizationService.canMakePaymentsResult = false
+    let sut = ApplePayModule(pkPaymentAuthorizationService: mockPaymentAuthorizationService)
+
+      // Then
+    sut.isAvailable(paymentMethodDict, configuration: configuration) { resolve in
+      let isSuccess = try! XCTUnwrap(resolve as? Bool )
+      XCTAssertFalse(isSuccess)
+    } rejecter: { title, message, error in
+      XCTFail("Shou not throw error")
+    }
+  }
+
+  func testApplePayDetailsExtraData() {
+      // When
+    let extraData = mockApplePayDetails.extraData
+    XCTAssertNotNil(extraData)
+  }
+
+    // Then
+  func testPKContactJsonObject() {
+      // Given
+    let contact = PKContact()
+    contact.emailAddress = "test@example.com"
+    contact.phoneNumber = CNPhoneNumber(stringValue: "+1234567890")
+
+    var name = PersonNameComponents()
+    name.givenName = "John"
+    name.familyName = "Doe"
+    contact.name = name
+
+      // When
+    let jsonObject = contact.jsonObject
+
+      // Then
+    XCTAssertNotNil(jsonObject)
+  }
+
+  func testPKPaymentNetworkTxVariantName() {
+      // Given/When/Then
+    XCTAssertEqual(PKPaymentNetwork.masterCard.txVariantName, "mc")
+    XCTAssertEqual(PKPaymentNetwork.cartesBancaires.txVariantName, "cartebancaire")
+    XCTAssertEqual(PKPaymentNetwork.visa.txVariantName, "visa")
+    XCTAssertEqual(PKPaymentNetwork.amex.txVariantName, "amex")
+  }
+
+  func testPKPaymentAuthorizationServiceAdapter() {
+      // Given
+    let adapter = PKPaymentAuthorizationServiceAdapter()
+
+      // When
+    let canMakePayments = adapter.canMakePayments(usingNetworks: [.visa, .masterCard])
+    let viewController = adapter.getAuthorizationViewController(paymentRequest: mockPaymentRequest)
+
+      // Then
+      // These methods delegate to the real PKPaymentAuthorizationViewController
+      // so we can't easily test the exact behavior without mocking the static methods
+      // But we can at least test that the methods don't crash and return expected types
+    XCTAssertNotNil(viewController)
+    XCTAssertTrue(canMakePayments)
```

Comment:
Is there a scenario when this can return false?

Reply:
Not on simulator. This is API-fixture test + allow us to test if Apple Pay API ever change